Start/End Indexes translated as negative values when using substring before/after transformation

Used gherkin scenario: 
  ``` When select "SubStringBefore" transformation
    And for "input-startIndex" input set "3"
    And for "input-match" input set "blah"
    And set source data
      | sourceString |
      | foobarblah   |
    And set expected data
      | targetString |
      | fo           |
    Then save and verify 
```
Following exception is thrown during processing in camel route: 

```java
 Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -3
	at java.lang.String.substring(String.java:1967)
	at io.atlasmap.actions.StringComplexFieldActions.doSubString(StringComplexFieldActions.java:316)
	at io.atlasmap.actions.StringComplexFieldActions.subStringBefore(StringComplexFieldActions.java:294)
	... 80 more

2018-01-08 14:35:49.309  INFO 55326 --- [       Thread-2] ationConfigEmbeddedWebApplicationContext : Closing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@4bf48f6: startup date [Mon Jan 08 14:35:40 CET 2018]; root of context hierarchy 
``` 

Created mappings.xml looks fine: 
```xml
<Mapping xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="Mapping" id="mapping.479590" mappingType="Map">
            <InputField xsi:type="ns2:JavaField" name="sourceString" docId="io.atlasmap.qe.test.SourceMappingTestClass" path="/sourceString" fieldType="String"/>
            <OutputField xsi:type="ns2:JavaField" name="targetString" docId="io.atlasmap.qe.test.TargetMappingTestClass" path="/targetString" fieldType="String">
                <Actions>
                    <SubStringBefore startIndex="3" endIndex="0" match="blah"/>
                </Actions>
            </OutputField>
        </Mapping>
    </Mappings>
```

I was trying to do https://github.com/atlasmap/atlasmap/blob/master/runtime/core/src/test/java/io/atlasmap/actions/StringComplexFieldActionsTest.java#L492 however I didn't realise that there is '0' as default. So I set the End index to '5'. But still I'm getting 
```
Caused by: java.lang.IllegalArgumentException: SubStringBefore action must be specified with a positive startIndex and a string to match
	at io.atlasmap.actions.StringComplexFieldActions.subStringBefore(StringComplexFieldActions.java:283)
	... 80 more
```

yes, but the another problem is that I'm not able to get this work. It ends at`Caused by: java.lang.IllegalArgumentException: SubStringBefore action must be specified with a positive startIndex and a string to match
	at io.atlasmap.actions.StringComplexFieldActions.subStringBefore(StringComplexFieldActions.java:283)
	... 80 more` even if I have indexes filled correctly as follows: 
startIndex = 3 
endIndex = 5 
match = 'blah'
input = 'foobarblah'. I'd expect to get substring not that exception

--- a/runtime/core/src/main/java/io/atlasmap/actions/StringComplexFieldActions.java
+++ b/runtime/core/src/main/java/io/atlasmap/actions/StringComplexFieldActions.java
@@ -288,7 +288,7 @@ public class StringComplexFieldActions implements AtlasFieldAction {
         if (action == null || !(action instanceof SubStringAfter) || ((SubStringAfter) action).getStartIndex() == null
                 || ((SubStringAfter) action).getStartIndex() < 0 || ((SubStringAfter) action).getMatch() == null
                 || (((SubStringAfter) action).getEndIndex() != null
-                        && ((SubStringAfter) action).getEndIndex() > ((SubStringAfter) action).getStartIndex())) {
+                        && ((SubStringAfter) action).getEndIndex() < ((SubStringAfter) action).getStartIndex())) {
             throw new IllegalArgumentException(
                     "SubStringAfter action must be specified with a positive startIndex and a string to match");
         }
@@ -312,7 +312,7 @@ public class StringComplexFieldActions implements AtlasFieldAction {
         if (action == null || !(action instanceof SubStringBefore) || ((SubStringBefore) action).getStartIndex() == null
                 || ((SubStringBefore) action).getStartIndex() < 0 || ((SubStringBefore) action).getMatch() == null
                 || (((SubStringBefore) action).getEndIndex() != null
-                        && ((SubStringBefore) action).getEndIndex() > ((SubStringBefore) action).getStartIndex())) {
+                        && ((SubStringBefore) action).getEndIndex() < ((SubStringBefore) action).getStartIndex())) {
             throw new IllegalArgumentException(
                     "SubStringBefore action must be specified with a positive startIndex and a string to match");
         }

--- a/runtime/core/src/test/java/io/atlasmap/actions/StringComplexFieldActionsTest.java
+++ b/runtime/core/src/test/java/io/atlasmap/actions/StringComplexFieldActionsTest.java
@@ -444,6 +444,12 @@ public class StringComplexFieldActionsTest {
 
         assertEquals("barblah", StringComplexFieldActions.subStringAfter(action, "barblah"));
 
+        action.setEndIndex(7);
+        assertEquals("blah", StringComplexFieldActions.subStringAfter(action, "foobarblahfoo"));
+
+        action.setEndIndex(3);
+        assertEquals("", StringComplexFieldActions.subStringAfter(action, "foobarblahfoo"));
+
         try {
             StringComplexFieldActions.subStringAfter(null, "aa");
             fail("IllegalArgumentException expected");
@@ -492,6 +498,11 @@ public class StringComplexFieldActionsTest {
         assertEquals("bar", StringComplexFieldActions.subStringBefore(action, "foobarblah"));
         assertEquals("foobar", StringComplexFieldActions.subStringBefore(action, "foofoobarblahfoo"));
         assertEquals("", StringComplexFieldActions.subStringBefore(action, "barblah"));
+
+        action.setEndIndex(5);
+        assertEquals("ba", StringComplexFieldActions.subStringBefore(action, "foobarblah"));
+        action.setEndIndex(3);
+        assertEquals("", StringComplexFieldActions.subStringBefore(action, "foobarblah"));
 
         try {
             StringComplexFieldActions.subStringBefore(null, "aa");

